addons/bus/static/src/services/bus_service.js

```diff
@@ -153,71 +146,41 @@ export const busService = {
                     session.websocket_worker_version
                 }`,
                 db: session.db,
                 debug: odoo.debug,
-                lastNotificationId: multiTab.getSharedValue("last_notification_id", 0),
+                lastNotificationId: legacyMultiTab.getSharedValue("last_notification_id", 0),
                 uid,
                 startTs: startedAt.valueOf(),
             });
         }
 
         /**
-         * Start the "bus_service" worker.
+         * Start the "bus_service" workerService.
          */
-        function startWorker() {
-            let workerURL = `${params.serverURL}/bus/websocket_worker_bundle?v=${session.websocket_worker_version}`;
-            if (params.serverURL !== window.origin) {
-                // Bus service is loaded from a different origin than the bundle
-                // URL. The Worker expects an URL from this origin, give it a base64
-                // URL that will then load the bundle via "importScripts" which
-                // allows cross origin.
-                const source = `importScripts("${workerURL}");`;
-                workerURL = "data:application/javascript;base64," + window.btoa(source);
-            }
-            const workerClass = isUsingSharedWorker ? browser.SharedWorker : browser.Worker;
-            worker = new workerClass(workerURL, {
-                name: isUsingSharedWorker
-                    ? "odoo:websocket_shared_worker"
-                    : "odoo:websocket_worker",
-            });
-            worker.addEventListener("error", (e) => {
-                if (!isInitialized && workerClass === browser.SharedWorker) {
-                    console.warn(
-                        'Error while loading "bus_service" SharedWorker, fallback on Worker.'
-                    );
-                    isUsingSharedWorker = false;
-                    startWorker();
-                } else if (!isInitialized) {
-                    isInitialized = true;
-                    connectionInitializedDeferred.resolve();
-                    console.warn("Bus service failed to initialized.");
-                }
-            });
-            if (isUsingSharedWorker) {
-                worker.port.start();
-                worker.port.addEventListener("message", handleMessage);
-            } else {
-                worker.addEventListener("message", handleMessage);
+        async function startWorker() {
+            if (!connectionInitializedDeferred) {
+                connectionInitializedDeferred = new Deferred();
             }
+            await workerService.registerHandler(handleMessage);
             initializeWorkerConnection();
         }
         browser.addEventListener("pagehide", ({ persisted }) => {
             if (!persisted) {
                 // Page is gonna be unloaded, disconnect this client
                 // from the worker.
-                send("BUS:LEAVE");
+                workerService.send("BUS:LEAVE");
```

Comment:
Hello, we actually noticed an issue here:

This might actually start the worker (in website in particular, where we don't start it by default) whenever we leave the page.
If the worker was not already started, we probably don't want to start it here just to send a STOP message.

@yhu-odoo can you look to have a fix?